DQMServices/Core/interface/DQMStore.h

```diff
@@ -189,440 +420,278 @@ namespace dqm::dqmstoreimpl {
-      MonitorElement* getElement(std::string const& path);
+      ~IGetter() override;
 
-      std::vector<std::string> getSubdirs();
-      std::vector<std::string> getMEs();
-      bool containsAnyMonitorable(std::string const& path);
-      bool dirExists(std::string const& path);
-      void cd();
-      void cd(std::string const& dir);
-      void setCurrentFolder(std::string const& fullpath);
+    protected:
+      IGetter(DQMStore* store);
 
-      IGetter() = delete;
-      IGetter(IGetter const&) = delete;
+      DQMStore* store_;
+    };
 
-    private:
-      explicit IGetter(DQMStore* store) noexcept : owner_{store} { assert(store); }
-
-      // Embedded classes do not natively own a pointer to the embedding
-      // class. We therefore need to store a pointer to the main
-      // DQMStore instance (owner_).
-      DQMStore* owner_;
-    };  //IGetter
-
-    // Template function to be used inside each DQM Modules' lambda
-    // functions to book MonitorElements into the DQMStore. The function
-    // calls whatever user-supplied code via the function f. The latter
-    // is passed the instance of the IBooker class (owned by the *only*
-    // DQMStore instance), that is capable of booking MonitorElements
-    // into the DQMStore via a public API. The central mutex is acquired
-    // *before* invoking and automatically released upon returns.
-    template <typename iFunc>
-    void bookTransaction(iFunc f, uint32_t run, uint32_t moduleId, bool canSaveByLumi) {
-      std::lock_guard<std::mutex> guard(book_mutex_);
-      /* Set the run number and module id only if multithreading is enabled */
-      if (enableMultiThread_) {
-        run_ = run;
-        moduleId_ = moduleId;
-        canSaveByLumi_ = canSaveByLumi;
-      }
-      IBooker booker{this};
-      f(booker);
-
-      /* Reset the run number and module id only if multithreading is enabled */
-      if (enableMultiThread_) {
-        run_ = 0;
-        moduleId_ = 0;
-        canSaveByLumi_ = false;
+    class DQMStore : public IGetter, public IBooker {
+    public:
+      // IGetter uses the globalMEs_ directly.
+      friend IGetter;
+      enum OpenRunDirs { KeepRunDirs, StripRunDirs };
+
+      DQMStore(edm::ParameterSet const& pset, edm::ActivityRegistry&);
+      ~DQMStore() override;
+
+      // ------------------------------------------------------------------------
+      // ---------------------- public I/O --------------------------------------
+      DQM_DEPRECATED
+      void save(std::string const& filename, std::string const& path = "");
+      DQM_DEPRECATED
+      bool open(std::string const& filename,
+                bool overwrite = false,
+                std::string const& path = "",
+                std::string const& prepend = "",
+                OpenRunDirs stripdirs = KeepRunDirs,
+                bool fileMustExist = true);
+
+      DQMStore(DQMStore const&) = delete;
+      DQMStore& operator=(DQMStore const&) = delete;
```

Comment:
It's a singleton (still `edm::Service`) for now, so that does not really apply.

It might actually be safe to move, but I'll rather just ban it since there is no need for moving.